pages/index.js

```diff
@@ -174,34 +192,43 @@ const ConfirmationDialog = styled.div`
   }
 `;
 
-function useLocalStorageState(key, defaultValue) {
-  const [state, setState] = useState(defaultValue);
-
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      const storedValue = localStorage.getItem(key);
-      if (storedValue) {
-        setState(JSON.parse(storedValue));
-      }
-    }
-  }, [key]);
-
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      localStorage.setItem(key, JSON.stringify(state));
-    }
-  }, [key, state]);
-
-  return [state, setState];
-}
-
 const ShoppingList = () => {
   const [items, setItems] = useLocalStorageState("shoppingItems", []);
+  const [purchasedItems, setPurchasedItems] = useLocalStorageState(
+    "purchasedItems",
+    []
+  );
   const [errors, setErrors] = useState({});
   const [deletingItemId, setDeletingItemId] = useState(null);
 
+  const totalItems = (items || []).reduce(
+    (total, item) => total + item.quantity,
+    0
+  );
+
+  const TotalPurchased = (purchasedItems || []).reduce(
+    (total, item) => total + item.quantity,
+    0
+  );
+  const handleCheckboxClick = (itemId) => {
+    const item = items.find((item) => item.id === itemId);
+    if (item) {
+      const purchasedItem = { ...item, purchased: true };
+      setPurchasedItems([purchasedItem, ...(purchasedItems || [])]);
+      setItems(items.filter((item) => item.id !== itemId));
+    }
+  };
+  const handleUnmarkClick = (itemId) => {
+    const purchasedItem = purchasedItems.find((item) => item.id === itemId);
+    if (purchasedItem) {
+      const item = { ...purchasedItem, purchased: false };
+      setItems([item, ...(items || [])]);
+      setPurchasedItems(purchasedItems.filter((item) => item.id !== itemId));
+    }
+  };
+
   useEffect(() => {
-    if (items.length === 0) {
+    if ((items || []).length === 0) {
```

Comment:
if items.length === 0 this means that items is an empty array so you don't need to use || here. items.length === 0 is enough